src/core/config/scripts/generateConfig.mjs

```diff
@@ -42,13 +42,32 @@ for (const opObj in Ops) {
         outputType:  op.presentType,
         flowControl: op.flowControl,
         manualBake:  op.manualBake,
-        args:        op.args
+        args:        op.args,
     };
 
-    if ("patterns" in op) {
-        operationConfig[op.name].patterns = op.patterns;
+    if ("checks" in op) {
+        if ("input" in op.checks) {
+            operationConfig[op.name].input = {};
```

Comment:
Just scanning this from a language perspective. You *could* use some [spread operator magic](https://medium.com/@mikeh91/conditionally-adding-keys-to-javascript-objects-using-spread-operators-and-short-circuit-evaluation-acf157488ede) to conditionally add properties to an object. I'll leave it up to you to decide which one is in fact more readable.

Reply:
Having taken a look at the spread operator method, I really don't like it. It has pretty ugly syntax and it's not immediately obvious how it works.

However, I definitely agree that conditional property access would tidy things up a lot. I think it's probably best to wait for [this TC39 proposal](https://github.com/tc39/proposal-optional-chaining) to work its way through though. It's a much cleaner solution.